### PR TITLE
Introduce new withdraw mechanism

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: contracts/truefi/TrueFiPool.sol [hunk 1]
```diff
@@ -257,6 +265,64 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
         emit Exited(msg.sender, amount);
     }
 
+    /**
+     * @dev Exit pool only with liquid tokens
+     * This function will withdraw TUSD but with a small penalty
+     * @param amount amount of pool tokens to redeem for underlying tokens
+     */
+    function liquidExit(uint256 amount) external nonReentrant {
+        require(amount <= balanceOf(msg.sender), "CurvePool: insufficient funds");
```
Comment (barrutko) [inline contracts/truefi/TrueFiPool.sol:274]:
```suggestion
        require(amount <= balanceOf(msg.sender), "CurvePool: Insufficient funds");
```

File: contracts/truefi/TrueFiPool.sol [hunk 2]
```diff
@@ -257,6 +265,64 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
...
+
+        uint256 amountToWithdraw = poolValue().mul(amount).div(totalSupply());
```
Comment (barrutko) [inline contracts/truefi/TrueFiPool.sol:276]: Maybe `valueToWithdraw`?

File: contracts/truefi/TrueFiPool.sol [hunk 3]
```diff
@@ -257,6 +265,64 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
...
+     */
+    function integrateAtPoint(uint256 x) public pure returns (uint256) {
+        return uint256(ABDKMath64x64.ln(ABDKMath64x64.fromUInt(x.add(50)))).mul(50000).div(2**64);
+    }
+
+    /**
+     * @dev Calculates average penalty on interfal [from; to]
+     */
+    function averageExitPenalty(uint256 from, uint256 to) public pure returns (uint256) {
+        require(from <= to, "CurvePool: to is before from");
```
Comment (barrutko) [inline contracts/truefi/TrueFiPool.sol:315]:
```suggestion
        require(from <= to, "CurvePool: Invalid range");
```

File: contracts/truefi/TrueFiPool.sol (hunk 2 above)
Comment (vanruch) [reply contracts/truefi/TrueFiPool.sol:276]: why? amount seems to suit better

Comment (barrutko) [reply contracts/truefi/TrueFiPool.sol:276]: It was a bit confusing at the beginning, but maybe you're right

**Review by barrutko — APPROVED** [review]

**Pull request merged**